src/mpid/ch4/netmod/ofi/ofi_send.h

```diff
@@ -336,25 +343,17 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_send_normal(const void *buf, MPI_Aint cou
         /* Store ordering unnecessary for dst_rank, so use relaxed store */
         MPL_atomic_relaxed_store_int(&MPIDI_OFI_REQUEST(sreq, util_id), dst_rank);
         match_bits |= MPIDI_OFI_HUGE_SEND;      /* Add the bit for a huge message */
+        /* Note: we are sending the ctrl header, rather than the actual data */
```

Comment:
So the idea here is to only send the control message and not the actual data in the first message? Could you combine the first message and the control message into a single buffer? That was one of the benefits of using the previous protocol. You could avoid an extra round trip for the first chunk of data.￼￼

Reply:
It complicates the mechanism and I don't see much benefit if we assume the performance is bandwidth dominant. Since you can't fit all data in the first message anyway, there is no saving of extra round trip.

Reply:
The data is directly received into the user buffer, so if we combine the control header and data, the control header is to be after the data or we'll need to move the data. Looking for a header after an arbitrary offset seems to be a weird pattern and I think it is better to avoid.

Reply:
That's fine then.